Web.md

```diff
@@ -124,6 +124,103 @@ MIME type mismatch or `opaque` response types
 `WebAssembly.CompileError`.
 
+### Structured serialization of a `WebAssembly.Memory`
+
+A `WebAssembly.Memory` is a
+[serializable object](https://html.spec.whatwg.org/multipage/infrastructure.html#serializable-objects), and will require changes to the
+[HTML Standard](https://html.spec.whatwg.org/multipage/infrastructure.html#serialization-steps) as follows:
+
+[StructuredSerializeInternal](https://html.spec.whatwg.org/multipage/infrastructure.html#structuredserializeinternal) adds a new step after step 12:
+
+* Otherwise, if *value* has a [[WebAssembly.Memory]] internal slot, then:
+  1. Let *size* be the number value.[[BufferObject]].[[ArrayBufferByteLength]].
+  2. If ! [IsSharedArrayBuffer](https://tc39.github.io/ecma262/#sec-issharedarraybuffer)(*value*.[[BufferObject]]) is true, then:
+    1. If *forStorage* is true, then throw a
+       "`[DataCloneError](https://heycam.github.io/webidl/#datacloneerror)`"
+       `[DOMException](https://heycam.github.io/webidl/#dfn-DOMException)`.
+    2. Otherwise:
+       1. Set *serialized* to { [[Type]]: "WebAssembly.Memory",
+          [[ArrayBufferData]]: value.[[BufferObject]].[[ArrayBufferData]],
+          [[ArrayBufferByteLength]]: size,
+          [[AgentCluster]]: the [current Realm Record](https://tc39.github.io/ecma262/#current-realm)'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) }.
+       2. Otherwise:
+         1. Let *dataCopy* be [CreateByteDataBlock](https://tc39.github.io/ecma262/#sec-createbytedatablock)(size).
+            NOTE: This can throw a RangeError exception on allocation failure.
+         2. Perform [CopyDataBlockBytes](https://tc39.github.io/ecma262/#sec-copydatablockbytes)
+            (dataCopy, 0,
+            value.[[BufferObject]].[[ArrayBufferData]], 0, size).
+         3. Set *serialized* to { [[Type]]: "WebAssembly.Memory",
+            [[ArrayBufferData]]: dataCopy, [[ArrayBufferByteLength]]: size }.
+
+[StructuredDeserialize](https://html.spec.whatwg.org/multipage/infrastructure.html#structureddeserialize) adds a new step after step 12:
+
+* Otherwise, if *serialized*.[[Type]] is "WebAssembly.Memory", then:
+  1. If *targetRealm*'s corresponding [agent cluster](https://tc39.github.io/ecma262/#sec-agent-clusters) is not
+     *serialized*.[[AgentCluster]], then throw a
+     "`[DataCloneError](https://heycam.github.io/webidl/#datacloneerror)`"
+     `[DOMException](https://heycam.github.io/webidl/#dfn-DOMException)`.
+  2. Otherwise,
+     if ! [IsSharedArrayBuffer](https://tc39.github.io/ecma262/#sec-issharedarraybuffer)(*serialized*.[[ArrayBufferData]]) is true, then:
+     Set *value* to a new WebAssembly.Memory object in *targetRealm*
+     whose [[BufferObject]] internal slot is set to a new
+     [[ArrayBufferData]] internal slot value is
+     set to a new `SharedArrayBuffer`, where:
+     1. The `SharedArrayBuffer`'s internal slot [[ArrayBufferData]]
```

Comment:
Nit: this should be consistified with the below; this says "The `SharedArrayBuffer`'s internal slot" whereas the below gives a name to it and says "_mem_'s internal slot"